cpp/log/logged_certificate.cc

```diff
@@ -0,0 +1,65 @@
+#include "log/logged_certificate.h"
+
+using ct::PreCert;
+using ct::PrecertChainEntry;
+using ct::SignedCertificateTimestamp;
+using ct::X509ChainEntry;
+
+namespace cert_trans {
+
+
+bool LoggedCertificate::CopyFromClientLogEntry(
+    const AsyncLogClient::Entry& entry) {
+  if (entry.leaf.timestamped_entry().entry_type() != ct::X509_ENTRY &&
+      entry.leaf.timestamped_entry().entry_type() != ct::PRECERT_ENTRY) {
+    LOG(INFO) << "unsupported entry_type: "
+              << entry.leaf.timestamped_entry().entry_type();
+    return false;
+  }
+
+  Clear();
+
+  ct::SignedCertificateTimestamp* const sct(mutable_contents()->mutable_sct());
+  sct->set_version(ct::V1);
+  sct->set_timestamp(entry.leaf.timestamped_entry().timestamp());
+  sct->set_extensions(entry.leaf.timestamped_entry().extensions());
+
+  mutable_contents()->mutable_entry()->set_type(
```

Comment:
How about doing the copy from `entry` thing and then filling in the missing bits from `leaf` rather than doing all of it manually? Might save on line count a bit.